scripts/dev/tidy.php

```diff
@@ -0,0 +1,394 @@
+    -f, --fix            Edit and save files in place
+    -q, --quiet          Do not output any message
+
+  EXAMPLES:
+
+    Display help: php tidy.php -h
+
+    Check php-src: scripts/dev/tidy.php .
+
+    Fix php-src: scripts/dev/tidy.php --fix --no-backup .
+
+    Fix custom path or file: ./tidy.php --fix input-path
+
+END;
+}
+
+/**
+ * Determine the script version. If script is located in the PHP repository the
+ * PHP version is returned, otherwise the script's file sha1 hash is returned.
+ */
+function version(): string
+{
+    $file = __DIR__.'/../../main/php_version.h';
+
+    if (file_exists($file)) {
+        $content = file_get_contents($file);
+        preg_match('/^\#define\ PHP\_VERSION\ \"([0-9\.\-a-z]+)\"$/m', $content, $version);
+
+        if (isset($version[1])) {
+            return $version[1];
+        }
+    }
+
+    return sha1_file(__FILE__);
+}
+
+/**
+ * Trim trailing spaces and tabs from each line.
+ */
+function trimTrailingWhitespace(string $content): string
+{
+    return preg_replace('/[\t ]+(\r?$)/m', '$1', $content);
+}
+
+/**
+ * Trim final new lines and append an EOL (end of line) based on the previous
+ * prevailing EOL type used LF (\n), CRLF (\r\n) or CR (\r).
+ */
+function trimFinalEol(string $content): string
+{
+    $trimmed = rtrim($content, " \t\r\n");
+    $trimmed .= getEol($trimmed);
+
+    return $trimmed;
+}
+
+/**
+ * Get end of line based on the prevailing LF, CRLF or CR new line characters.
+ */
+function getEol(string $content): ?string
+{
+    // Empty files don't need a new line attached
+    if (empty($content)) {
+        return null;
+    }
+
+    // Match all LF, CRLF and CR EOL characters
+    preg_match_all('/(*BSR_ANYCRLF)\R/', $content, $matches);
+
+    // For single line files LF is returned
+    if (empty($matches[0])) {
+        return "\n";
+    }
+
+    $counts = array_count_values($matches[0]);
+    arsort($counts);
+    $prevailingEol = key($counts);
+
+    return $prevailingEol;
+}
+
+$fix = false;
+$whitespace = true;
+$eol = true;
+$colors = true;
+$backup = true;
+$quiet = false;
+
+foreach ($argv as $argument) {
+    switch ($argument) {
+        case ($argc < 2):
+        case '-h':
+        case '--help':
+            echo usage();
+            exit;
+        break;
+        case '--no-whitespace':
+            $whitespace = false;
+        break;
+        case '--no-eol':
+            $eol = false;
+        break;
+        case '-f':
+        case '--fix':
+            $fix = true;
+        break;
+        case '--no-colors':
+            $colors = false;
+        break;
+        case '--no-backup':
+            $backup = false;
+        break;
+        case '-q':
+        case '--quiet':
+            $quiet = true;
+        break;
+        case '-v':
+        case '--version':
+            echo "PHP tidy script version ".version()."\n";
+            exit;
+        break;
+        case (preg_match('/^-.+/', $argument) ? true : false):
```

Comment:
This is still using conditions inside switch. They work by accident, because for most values of `$argument`, `$argument == true` is true. This style of code will fail if for example `$argument == "0"`.

Instead these should be simple `if` statements inside the `default` clause.

Reply:
Got it. Added to the default case with if statement instead.